packages/gatsby-image/src/index.js

```diff
@@ -48,18 +48,54 @@ const convertProps = props => {
   return convertedProps
 }
 
+/**
+ * Checks if fluid or fixed are art-direction arrays.
+ *
+ * @param currentData  {array}   The props to check for images.
```

Comment:
The `array` and `object` types don't add much value in terms of typing. They might as well be `any` or `*` here. So please be specific in what the input type is here (see below)